yugabyte/src/yugabyte/ysql/client.clj

```diff
@@ -206,3 +218,99 @@
   "Like jdbc execute!!, but includes a default timeout."
   [conn sql-params]
   (j/execute! conn sql-params {:timeout default-timeout}))
+
+; Should probably be the last definition in order to have access to everything else
+(defmacro defclient
+  "Helper for defining YSQL clients.
+  Takes a new class name symbol and a definition of YSQLClientBase record
+  whose methods will be wrapped and called.
+  This approach is (arguably) cleaner than the one used for YCQL defclient.
```

Comment:
Please add a description that we only need separate `defrecord` in this approach in order to have better reporting in case of possible compilation errors during further development.